spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSqlExtensionsAstBuilder.scala

```diff
@@ -81,6 +87,85 @@ class IcebergSqlExtensionsAstBuilder(delegate: ParserInterface) extends IcebergS
       Option(ctx.name).map(_.getText))
   }
 
+  /**
+   * Create an ADD BRANCH logical command.
+   */
+  override def visitCreateBranch(ctx: CreateBranchContext): CreateBranch = withOrigin(ctx) {
+    val snapshotRetention = Option(ctx.snapshotRetentionClause())
+
+    CreateBranch(
+      typedVisit[Seq[String]](ctx.multipartIdentifier),
+      ctx.identifier().getText,
+      Option(ctx.snapshotId()).map(_.getText.toLong),
+      snapshotRetention.flatMap(s => Option(s.numSnapshots())).map(_.getText.toLong),
+      snapshotRetention.flatMap(s => Option(s.snapshotRetain())).map(
+        _.getText.toLong * timeUnit(ctx.snapshotRetentionClause().snapshotRetainTimeUnit().getText)),
```

Comment:
What will happen here if the `snapshotRetentionTimeUnit()` doesn’t resolve?

Reply:
The `snapshotRetentionTimeUnit` and `snapshotRetain` are used together. If `snapshotRetain` is unavailable, the `flatMap` will return a `None`, so the `map` does not execute.